backend/middlewares/auth-middleware.js

```diff
@@ -2,6 +2,7 @@ import { JWT_SECRET } from '../config/utils.js';
 import { ApiError } from '../utils/api-error.js';
 import { HTTP_STATUS, RESPONSE_MESSAGES } from '../utils/constants.js';
 import jwt from 'jsonwebtoken';
+import { Role } from '../types/role-type.js';
 
```

Comment:
Are you sure this is not giving any compilation issues? role-type i see is of tsx and here it is imported as js